src/java.base/share/classes/java/io/ByteArrayInputStream.java

```diff
@@ -207,10 +207,26 @@ public int readNBytes(byte[] b, int off, int len) {
     public synchronized long transferTo(OutputStream out) throws IOException {
         int len = count - pos;
         if (len > 0) {
+            // 'tmp' is null if and only if 'out' is trusted
+            byte[] tmp;
+            Class<?> outClass = out.getClass();
+            if (outClass.getPackageName().equals("java.io") &&
```

Comment:
For what do we need this string-based check here?

Reply:
I suspect it's left over from a previous iteration. In any case, limiting it to a small number of output streams makes this easier to look at. BAOS and FOS seem okay, POP seems okay too but legacy and not interesting.

Reply:
> I suspect it's left over from a previous iteration. In any case, limiting it to a small number of output streams makes this easier to look at. BAOS and FOS seem okay, POP seems okay too but legacy and not interesting.

Agreed for a rather short list of explicitly whitelisted implementations. We should get rid of the package check.

Reply:
I checked all the `OutputStreams` in the list for trustworthiness. The package check is vestigial; will remove. It could be useful if multiple packages were involved with multiple trusted classes in each.